tests/test_dataset.py

```diff
@@ -876,13 +893,15 @@ def test_current_batch_index(self):
         torch.manual_seed(42)
 
         class TestTaskEncoder(TaskEncoder):
```

Comment:
TODO: Add test: Save the random numbers once, then every run the numbers must be the same, otherwise we have a breaking change if randomness across runs changes